### DISCUSSION TITLE
train slowly，gpu-util 100% ，but pow 99w/450w

### DISCUSSION THREAD

**Discussion by qwerdf4** [huggingface/diffusers#7121] [Q&A] [answered]
accelerate launch .\train_dreambooth_lora_sdxl.py , my 4090 gpu-util 100% ，but pow only 99w/450w,  gpu fan: 30%,  3000steps need almost 40hours.

**Comment by asomoza** [comment, marked as answer]
you'll need to provide more information (full command for training), if I have to guess you probably have a high image batch number and you're using all the VRAM and started using the RAM.

**Reply by qwerdf4** [reply]
accelerate launch .\train_dreambooth_lora_sdxl.py --mixed_precision='fp16' --pretrained_model_name_or_path="" --dataset_name="" --instance_prompt='' --validation_prompt="" --output_dir="" --caption_column="text" --resolution=1024 --report_to="wandb" --snr_gamma=5.0 --checkpointing_steps=100 --train_batch_size=4 --gradient_accumulation_steps=1 --learning_rate=1e-4 --lr_scheduler="constant" --lr_warmup_steps=0 --max_train_steps=3000 --optimizer="adamw"

**Reply by qwerdf4** [reply]
![image](https://github.com/huggingface/diffusers/assets/57136063/f7759b57-b5de-421a-8682-0eba29244c96)
![image](https://github.com/huggingface/diffusers/assets/57136063/190ed1d7-6672-4d19-ab66-6fd0e5778886)

**Reply by qwerdf4** [reply]
the gpu-util always 100%，but power always lowly

**Reply by asomoza** [reply]
yeah, you're using all the GPU VRAM and using shared RAM.

With 24 GB VRAM you'll need a couple of params to start and then you can try changing them:

- you're missing `--gradient_checkpointing`
- use `--use_8bit_adam`
- use `--train_batch_size=1` and then start going higher.

Always watch the VRAM of your GPU, in your case never go higher than 23.4 GB or it will start using RAM and it's going to be slow.

**Reply by qwerdf4** [reply]
![image](https://github.com/huggingface/diffusers/assets/57136063/620d32c3-73e8-4e3c-a877-c4f9b25fda69)
it seems that bitsandbytes not friendly for Windows ，can't use 8 bit adam
![image](https://github.com/huggingface/diffusers/assets/57136063/4a61431b-fc55-4e1d-9267-77013fcff677)

**Reply by asomoza** [reply]
I forgot, yes, the bitsandbytes is not that easy to install but also not that complicated. you can try with this command:

`pip install bitsandbytes --prefer-binary --extra-index-url=https://jllllll.github.io/bitsandbytes-windows-webui`

if that doesn't work, you can manually download the wheel from here:

https://github.com/jllllll/bitsandbytes-windows-webui/releases/tag/wheels

I have tested this one:

https://github.com/jllllll/bitsandbytes-windows-webui/releases/download/wheels/bitsandbytes-0.41.1-py3-none-win_amd64.whl

ant then install it directly, like this:

`pip install bitsandbytes-0.41.1-py3-none-win_amd64.whl`

**Reply by qwerdf4** [reply]
Thank you for your reply! which helped me solve this problem! my model is training quickly now!
![image](https://github.com/huggingface/diffusers/assets/57136063/96348cb8-3205-4e29-bb2c-d313f1fba05c)

**Reply by qwerdf4** [reply]
Okay, I'll try it !
